python/pyspark/ml/torch/tests/test_distributor.py

```diff
@@ -122,6 +122,43 @@ def train_fn(learning_rate: float) -> Any:
     return train_fn
 
 
+def set_up_test_dirs():
+    gpu_discovery_script_file = tempfile.NamedTemporaryFile(delete=False)
+    gpu_discovery_script_file_name = gpu_discovery_script_file.name
+    gpu_discovery_script_file.write(
+        b'echo {\\"name\\": \\"gpu\\", \\"addresses\\": [\\"0\\",\\"1\\",\\"2\\"]}'
```

Comment:
no biggie but I would convert a dict to JSON and write it after encoding it.

Reply:
I don't find a easy way to add the `\`, let me just follow existing test for now